web/web.go

```diff
@@ -0,0 +1,1022 @@
+			err, status := canRenameOrChangeType(newColName, table)
+			if err != nil {
+				http.Error(w, fmt.Sprintf("%v", err), status)
+				return
+			}
+			updateType(v.ToType, table, newColName, srcTableName, w)
+		}
+		if v.NotNull != "" {
+			updateNotNull(v.NotNull, table, newColName)
+		}
+	}
+	app.conv.AddPrimaryKeys()
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(app.conv)
+}
+
+// getConversionRate returns table wise color coded conversion rate.
+func getConversionRate(w http.ResponseWriter, r *http.Request) {
+	reports := internal.AnalyzeTables(app.conv, nil)
+	rate := make(map[string]string)
+	for _, t := range reports {
+		rate[t.SpTable] = rateSchema(t.Cols, t.Warnings, t.SyntheticPKey != "")
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(rate)
+}
+
+// getSchemaFile generates schema file and returns file path.
+func getSchemaFile(w http.ResponseWriter, r *http.Request) {
+	ioHelper := &conversion.IOStreams{In: os.Stdin, Out: os.Stdout}
+	var err error
+	now := time.Now()
+	filePrefix, err := getFilePrefix(now)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Can not get file prefix : %v", err), http.StatusInternalServerError)
+	}
+	schemaFileName := "frontend/" + filePrefix + "schema.txt"
+	conversion.WriteSchemaFile(app.conv, now, schemaFileName, ioHelper.Out)
+	schemaAbsPath, err := filepath.Abs(schemaFileName)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Can not create absolute path : %v", err), http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte(schemaAbsPath))
+}
+
+// getReportFile generates report file and returns file path.
+func getReportFile(w http.ResponseWriter, r *http.Request) {
+	ioHelper := &conversion.IOStreams{In: os.Stdin, Out: os.Stdout}
+	var err error
+	now := time.Now()
+	filePrefix, err := getFilePrefix(now)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Can not get file prefix : %v", err), http.StatusInternalServerError)
+	}
+	reportFileName := "frontend/" + filePrefix + "report.txt"
+	conversion.Report(app.driver, nil, ioHelper.BytesRead, "", app.conv, reportFileName, ioHelper.Out)
+	reportAbsPath, err := filepath.Abs(reportFileName)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Can not create absolute path : %v", err), http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte(reportAbsPath))
+}
+
+type TableInterleaveStatus struct {
+	Possible bool
+	Parent   string
+	Comment  string
+}
+
+// setParentTable checks whether specified table can be
+// interleaved, if yes then it sets the Parent table for the specified
+// table and returns Parent table name, otherwise returns the issue.
+func setParentTable(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	if app.conv == nil || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
```

Comment:
s/spanner/Spanner/ in error message.

Reply:
changed to
Schema is not converted or Driver is not configured properly. Please retry converting the database to Spanner.